### PR TITLE
Fix: Ensure almost reduced values in assertOnCurve for ECDSA compliance

### PR DESCRIPTION
**Pull request by MartinMinkov**
This should probably be targetting v2

### CODE REVIEW

File: src/lib/provable/gadgets/elliptic-curve.ts [hunk 1]
```diff
@@ -166,7 +166,9 @@ function assertOnCurve(
 ) {
   let { x, y } = p;
   let x2 = ForeignField.mul(x, x, f);
+  ForeignField.assertAlmostReduced([x2], f);
   let y2 = ForeignField.mul(y, y, f);
+  ForeignField.assertAlmostReduced([y2], f);
```
[the remark this answers is not in the document]
Comment (mitschabaude) [reply src/lib/provable/gadgets/elliptic-curve.ts:171]: if we make those changes, we can remove the `assertAlmostReduced()` call in `ForeignCurveV2.check()`

[the remark this answers is not in the document]
Comment (MartinMinkov) [reply src/lib/provable/gadgets/elliptic-curve.ts]: Awesome! Made those changes, thanks!

**Review by mitschabaude — APPROVED** [review]

**Comment by mitschabaude** [comment]
hey @MartinMinkov would be ideal to merge `main` into `v2`, and `v2` into this branch, before merging, so that this is easier to review for auditors (right now the PR diff shows has a bunch of unrelated changes)

**Comment by MartinMinkov** [comment]
> hey @MartinMinkov would be ideal to merge main into v2, and v2 into this branch, before merging, so that this is easier to review for auditors (right now the PR diff shows has a bunch of unrelated changes)

Yep, sounds good! I have https://github.com/o1-labs/o1js/pull/1744 that I want to merge asap to fix this

**Pull request merged**